### ISSUE TITLE
False negative: no errors when statement always false due to incompatible types

### ISSUE THREAD

**Issue by dhorkin** [python/mypy#7911]
Toy example:
```Python
bar= "FOO"

if bar.lower in ['foo']:
    print("foo")

arr: List[str] = ['foo']
if bar.lower in arr:
    print("foo")
```

As lower is missing the (), it's actually a function, not a string. As a result, given the right-hand side is a `List[str]`, I would have expected mypy to give an error, as regardless of what `foo` may be. The fact the statement will always resolve to false due to the incompatible types ought to flag an error since it's almost certainly a bug (I have mypy 0.730 and python 3.7)

Instead, mypy reports zero errors. Even when I explicitly declare the list as a list of strings, it doesn't seem to notice the issue.

**Comment by Michael0x2a** [comment]
I think the [`--strict-equality`](https://mypy.readthedocs.io/en/latest/command_line.html#cmdoption-mypy-strict-equality) flag is what you want here. When that flag is enabled, mypy will report the following error on both if statements:

```
Non-overlapping container check (element type: "Callable[[], str]", container item type: "str")
```
